### PR TITLE
Infra: Add teams-provider project skeleton.

### PR DESCRIPTION
**Pull request by wildeyes**
Need help, please take a look.

Running from docker-compose, and going to `/teams/1` will produce a page that says `id: {}` while starting the project directly, going to the same url will produce a page that says `id: { id: 1 }`.

The dynamic param is not being routed correctly when using next under docker, and I can't figure out why, will love some help with this.

### CODE REVIEW

**Comment by wildeyes** [comment]
I'm seeing an error in the console that has to be relevant:
`GET http://localhost:2015/_next/static/development/pages/team/[teamid].js?ts=1580574216184 net::ERR_ABORTED 404 (Not Found)`

**Comment by wildeyes** [comment]
I've managed

File: providers/teams/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+FROM node:10
+
+# Create app directory
+WORKDIR /usr/src/app
+
+# Install app dependencies
+# A wildcard is used to ensure both package.json AND package-lock.json are copied
+# where available (npm@5+)
+COPY package*.json ./
```
Comment (shakedlokits) [inline providers/teams/Dockerfile:9]: If you're binding the entire volume then there's no need for the `COPY` layer

File: providers/teams/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+RUN npm install
+# If you are building your code for production
+# RUN npm ci --only=production
+
+# Bundle app source
+COPY . .
```
Comment (shakedlokits) [inline providers/teams/Dockerfile:16]: Here as well

File: providers/teams/lib/auth.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import cookie from 'js-cookie'
+import nextCookie from 'next-cookies'
+import Router from 'next/router'
+import { gql } from 'apollo-boost'
+import { apolloClient } from './apollo'
+import { decode } from 'jsonwebtoken'
+
+export const withAuthentication = WrappedComponent => {
```
Comment (shakedlokits) [inline providers/teams/lib/auth.tsx]: Let's talk F2F about this 👍

File: providers/teams/lib/common.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    id: number
+    teamId: number // FK
+    challengeId: number // FK
+}
+export interface Team_Members {
+    id: number
+    teamId: number // FK
+    userId: number // FK
+}
+export interface Challenge {
```
Comment (shakedlokits) [inline providers/teams/lib/common.ts]: Commented on this in design.

File: providers/teams/pages/team/[teamid]/index.tsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    const router = useRouter()
+    const { teamid: teamId = 10 } = router.query
+    const { loading, error, data } = useQuery(GET_TEAM, {
+        variables: { teamId },
+    });
+    
+}
+
+
+export default dynamic(
```
Comment (shakedlokits) [inline providers/teams/pages/team/[teamid]/index.tsx]: Nice!! 💯

File: services/teams/resolvers.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+}
+interface Challenge {
+    id: number
+    description: String
+    createdByUserId: number // FK, Defines whether it's a team project (created by a user of the project) or sponsered challenge (created by a "user" of the sponser)
+}
+
+export default {
+    Query: {
+        register: async (parent, { id }) => {
```
Comment (shakedlokits) [inline services/teams/resolvers.ts]: Not seeing any query here that would require this over Hasura pg schemas.

**Review by shakedlokits — CHANGES_REQUESTED** [review]

File: services/teams/resolvers.ts (hunk 6 above)
Comment (wildeyes) [reply services/teams/resolvers.ts]: Removed team service.

File: providers/teams/pages/team/[teamid]/index.tsx (hunk 5 above)
Comment (wildeyes) [reply providers/teams/pages/team/[teamid]/index.tsx]: thanks.

File: providers/teams/Dockerfile (hunk 1 above)
Comment (wildeyes) [reply providers/teams/Dockerfile:9]: I'll keep until there's infra for dev mode with docker (as per our talk)

File: Caddyfile [hunk 7]
```diff
@@ -1,5 +1,6 @@
 0.0.0.0
 proxy / registration-provider:3000
+proxy /team teams-provider:3000
```
Comment (shakedlokits) [inline Caddyfile:3]: Not blocking but could probably be `/teams` pluraly because it's the `teams provider`

**Review by shakedlokits — APPROVED** [review]
Very nice, good job! 👍

**Comment by wildeyes** [comment]
I see where you are coming from, but I must suggest that your looking from
a technical perspective. The UTL exposed to the user concerns with user
experience, and from the users perspective he’s dealing with _a_ team.

/team/1
/team/create
/team/search
/team/1/edit

Make less sense when it’s “teams”

**Pull request merged**